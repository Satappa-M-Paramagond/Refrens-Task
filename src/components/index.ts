export { FormatText } from './FormatText';
export { Loader } from './Loader';
export { PrimaryButton } from './PrimaryButton';
export { UserProfileCard } from './UserProfileCard';
