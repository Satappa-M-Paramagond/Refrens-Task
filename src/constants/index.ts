export { Colors } from './Colors';
export { GetIDFromURL, TestIDs } from './Constants';
export { Scale } from './Scale';
export { Screens } from './Screens';
export { Strings } from './Strings';
