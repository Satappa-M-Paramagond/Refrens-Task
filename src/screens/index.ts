export { HomeController } from './home';
export { ProfileController } from './profile';
export { ResidentsController } from './residents';
