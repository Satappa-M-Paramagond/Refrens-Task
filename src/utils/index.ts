export { Axios, ApiMethods, ApiPaths, ApiService } from './ApiService';
