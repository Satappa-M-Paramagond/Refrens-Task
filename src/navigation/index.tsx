import React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import {
    HomeController,
    ProfileController,
    ResidentsController
} from '../screens';
import { Colors, Screens } from '../constants';
import { StatusBar } from 'react-native';

const HomeStack = createNativeStackNavigator();

export const AppNavigator = () => {
    return (
        <NavigationContainer>
            <StatusBar
                barStyle={'dark-content'}
                backgroundColor={Colors.transparent}
                translucent={true}
            />
            <HomeStack.Navigator>
                <HomeStack.Screen
                    name={Screens.HOME}
                    component={HomeController}
                />
                <HomeStack.Screen
                    name={Screens.PROFILE}
                    component={ProfileController}
                />
                <HomeStack.Screen
                    name={Screens.RESIDENTS}
                    component={ResidentsController}
                />
            </HomeStack.Navigator>
        </NavigationContainer>
    );
};
