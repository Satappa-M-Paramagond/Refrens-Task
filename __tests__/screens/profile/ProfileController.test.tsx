jest.useFakeTimers();
import React from 'react';
import renderer from 'react-test-renderer';
import { ProfileController } from '../../../src/screens';
import { mount } from 'enzyme';
import {
    episodesData,
    locationsData,
    usersData
} from '../../__mock__/mockData';

const props = {
    route: {
        params: {
            characterDetails: usersData[0]
        }
    }
};

describe('ProfileController => Snapshots => ', () => {
    test('should render loading view properly', () => {
        const wrapper = mount(<ProfileController {...props} />);
        const instance = wrapper.instance() as ProfileController;
        instance.setState({
            isLoading: true
        });
        wrapper.update();

        const tree = renderer.create(<ProfileController {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
        expect.assertions(1);
    });

    test('should render empty view properly', () => {
        const wrapper = mount(<ProfileController {...props} />);
        const instance = wrapper.instance() as ProfileController;
        instance.setState({
            isLoading: false,
            characterDetails: usersData[0],
            originInfo: {},
            locationInfo: {},
            episodesList: []
        });
        wrapper.update();

        const tree = renderer.create(<ProfileController {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
        expect.assertions(1);
    });

    test('should render list view properly', () => {
        const wrapper = mount(<ProfileController {...props} />);
        const instance = wrapper.instance() as ProfileController;
        instance.setState({
            isLoading: false,
            characterDetails: usersData[0],
            originInfo: locationsData[0],
            locationInfo: locationsData[1],
            episodesList: episodesData
        });
        wrapper.update();

        const tree = renderer.create(<ProfileController {...props} />).toJSON();
        expect(tree).toMatchSnapshot();
        expect.assertions(1);
    });
});
