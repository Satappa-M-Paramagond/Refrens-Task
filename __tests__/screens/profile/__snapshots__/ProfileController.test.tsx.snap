// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileController => Snapshots =>  should render empty view properly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <ActivityIndicator
    color="darkgrey"
    size="large"
  />
</View>
`;

exports[`ProfileController => Snapshots =>  should render list view properly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <ActivityIndicator
    color="darkgrey"
    size="large"
  />
</View>
`;

exports[`ProfileController => Snapshots =>  should render loading view properly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      },
      Object {},
    ]
  }
>
  <ActivityIndicator
    color="darkgrey"
    size="large"
  />
</View>
`;
